internal/alpha/printers/table/printer.go

```diff
@@ -72,87 +75,23 @@ func (t *Printer) Print(ch <-chan event.Event, _ common.DryRunStrategy, _ bool)
 }
 
 // columns defines the columns we want to print
-//TODO: We should have the number of columns and their widths be
+// TODO: We should have the number of columns and their widths be
 // dependent on the space available.
 var (
-	actionColumnDef = table.ColumnDef{
-		// Column containing the resource type and name. Currently it does not
-		// print group or version since those are rarely needed to uniquely
-		// distinguish two resources from each other. Just name and kind should
-		// be enough in almost all cases and saves space in the output.
-		ColumnName:   "action",
-		ColumnHeader: "ACTION",
-		ColumnWidth:  12,
-		PrintResourceFunc: func(w io.Writer, width int, r table.Resource) (int,
-			error) {
-			var resInfo *resourceInfo
-			switch res := r.(type) {
-			case *resourceInfo:
-				resInfo = res
-			default:
-				return 0, nil
-			}
-
-			var text string
-			switch resInfo.ResourceAction {
-			case event.ApplyAction:
-				if resInfo.ApplyStatus != event.ApplyFailed {
-					text = resInfo.ApplyStatus.String()
-				}
-			case event.PruneAction:
-				if resInfo.PruneStatus != event.PruneFailed {
-					text = resInfo.PruneStatus.String()
-				}
-			}
-
-			if len(text) > width {
-				text = text[:width]
-			}
-			_, err := fmt.Fprint(w, text)
-			return len(text), err
-		},
-	}
-
-	reconciledColumnDef = table.ColumnDef{
-		// Column containing the reconciliation status.
-		ColumnName:   "reconciled",
-		ColumnHeader: "RECONCILED",
-		ColumnWidth:  10,
-		PrintResourceFunc: func(w io.Writer, width int, r table.Resource) (
-			int,
-			error,
-		) {
-			var resInfo *resourceInfo
-			switch res := r.(type) {
-			case *resourceInfo:
-				resInfo = res
-			default:
-				return 0, nil
-			}
-
-			var text string
-			switch resInfo.ResourceAction {
-			case event.WaitAction:
-				text = resInfo.WaitStatus.String()
-			}
-
-			if len(text) > width {
-				text = text[:width]
-			}
-			_, err := fmt.Fprint(w, text)
-			return len(text), err
-		},
+	unifiedStatusColumnDef = table.ColumnDef{
+		// Column containing the overall progress.
+		ColumnName:        "progress",
+		ColumnHeader:      "PROGRESS",
+		ColumnWidth:       80,
+		PrintResourceFunc: printProgress,
 	}
 
-	columns = []table.ColumnDefinition{
+	alphaColumns = []table.ColumnDefinition{
 		table.MustColumn("namespace"),
 		table.MustColumn("resource"),
-		actionColumnDef,
-		table.MustColumn("status"),
-		reconciledColumnDef,
-		table.MustColumn("conditions"),
-		table.MustColumn("age"),
-		table.MustColumn("message"),
+
+		// We are trying out a "single column" model here
+		unifiedStatusColumnDef,
```

Comment:
Does this change applies only to `kpt alpha live apply ?` or `kpt live apply` as well.

If this applies to `kpt alpha` only, then we don't have to think much about any surprises this may cause to the end users. Thinking more, since `--output=table` is not default, so maybe it is ok.

Reply:
Only alpha live apply (the previous commit split them), and only --output=table

I wonder if we should also make table the default for apply (alpha initially) and maybe also server-side the default also.  We can also make the default switch based on whether the output is a TTY or not (I think).  But those changes would be in future PRs!

Reply:
experimenting with the UX in `alpha` makes sense. @mortent raises some very good point about why we have these additional fields.

FWIW, I found `--output=table` to be more useful and I end up specifying that always, so +1 to making it default.